tests/xharness/BCLTestImporter/BCLTestImportTargetFactory.cs

```diff
@@ -27,6 +31,29 @@ public BCLTestImportTargetFactory (Harness harness)
 			};
 		}
 		
+		async Task<TestExecutingResult> RestoreNugetsAsync (string projectPath)
```

Comment:
We already have something similar here: https://github.com/xamarin/xamarin-macios/blob/c28937e502d27ee4f999263d642bf34f224ef1a7/tests/xharness/Jenkins.cs#L2483-L2511

so we should try to reuse that somehow (it will probably require some modifications; that's fine).

Reply:
I used that as a base, I'll move it a little so we can use it :)